### ISSUE TITLE
Show only one feed entry per document

### ISSUE THREAD

**Issue by asaunier** [c2corg/v6_api#598]
Moved from the UI issue https://github.com/c2corg/v6_ui/issues/1164

**Summary:**

Documents advertised in the feed should be advertised only once (= only 1 related card), whether for the document creation, images uploading or users associations.

If a new event is recorded for a given document (for instance an associated user has uploaded images to an outing), only this last even should be mentioned in the feed (=have a card displayed). All previous events (creation, associated users) should no longer be mentioned (no card).

**Comment by asaunier** [comment]
@tsauerwein I see that when you update the title of an outing, this title is updated in the feed card as well (because a join is made to get the document title using its document id?).
What happens if someone updates the document quality? Could it also be updated in the feed cards?
If yes we could actually use it to filter feed cards, only making cards with some quality appear, coud we?

**Comment by tsauerwein** [comment]
When you request the feed, feed entries are first selected from `feed_document_changes`. Then the documents are loaded for these entries (if possible from the cache). That's why you get the updated name.

If you want to select feed entries depending on the document quality, you would have to store the quality in `feed_document_changes` and update if the field is changed.

**Comment by asaunier** [comment]
> If you want to select feed entries depending on the document quality, you would have to store the quality in feed_document_changes and update if the field is changed.

OK thanks. That's probably a change that makes sense: there's a great demand from the client to remove poor documents from the feed. But we should also manage the case where people have set an incorrect quality (or have improved the document - for instance their outing) and update it later => even though we don't advertise the document update, taking the new quality into account would make sense.

**Comment by asaunier** [comment]
@desnoes @stef74 

@tsauerwein just made me realize that the "one document = one card in the feed" constraint has an annoying impact on the profile feed.

Here is a use case:
* user A creates a route and upload images to it => both the home feed and user A's profile feed will show a card saying that "user A has uploaded images to this route".
* 1h later, user B uploads images to the same route => as expected only the last images upload would be shown in the home feed ("user B has uploaded images to this route"). The card will also appear in user B's profile feed. But since we only have one card per document, user B's change card will replace user A's => the route card will no longer appear in user A's profile feed.

This means contributions of a given user might disappear from the user's feed if someone else uploads images on the same documents in the meantime. Outing cards will remain though because of the participants association mechanism (but not available for other document types).

Is it OK for you?

@tsauerwein has suggested another approach: "one card per document and per user".
Code change is available at https://github.com/c2corg/v6_api/commit/2c50e2d31fa8b4dfcf4d827c0b24d4b1b1e60221
=> for a given document, a single card is shown for all actions of a user (eg. if the user creates an outing and upload 10 images separately) but if another user also uploads images (including 10 images separately) a another card will be shown as well.

Since cards are not shared among several users, the cards will remain shown in the profile feeds of those users, even though someone else uploads other images in the meantime.

What approach should be used? Thanks!

**Comment by desnoes** [comment]
> @tsauerwein has suggested another approach: "one card per document and per user".
@asaunier : it's ok for me.

**Comment by desnoes** [comment]
ok on demo
